Stack corruption in TColorGradient on mac beta

### Check duplicate issues.

- [x] Checked for duplicates

### Description

The tutorials intermittently crash as follows:
```
 1998/3684 Test #1158: tutorial-visualisation-gl-grad2 ...................................................................***Failed    0.54 sec
  Processing /Users/sftnight/ROOT-CI/src/tutorials/visualisation/gl/grad2.C...
  CMake Error at /Users/sftnight/ROOT-CI/src/cmake/modules/RootTestDriver.cmake:253 (message):
    error code: Subprocess aborted
```

Attaching lldb, this is due to a corruption of the stack canary when `-O2` is used.

### Reproducer

Nightlies

### ROOT version

master

### Installation method

nightlies

### Operating system

mac beta

### Additional context

_No response_

--- a/core/base/src/TColorGradient.cxx
+++ b/core/base/src/TColorGradient.cxx
@@ -111,6 +111,12 @@ void TColorGradient::ResetColor(UInt_t nPoints, const Double_t *points,
    fColors.assign(colors, colors + nPoints * 4);
 
    Double_t sum[4] = { 0., 0., 0., 0. };
+#ifdef __clang__
+   // Clang vectoriser bug (confirmed for clang 20 - 22)
+   // https://github.com/llvm/llvm-project/issues/194008
+   // The store in the loop can smash the stack with e.g. -O2 -mavx2.
+#pragma clang loop vectorize(disable)
+#endif
    for (unsigned n = 0; n < fColors.size(); ++n)
       sum[n % 4] += fColors[n];
 

--- a/core/base/test/CMakeLists.txt
+++ b/core/base/test/CMakeLists.txt
@@ -29,25 +29,6 @@ ROOT_ADD_GTEST(CoreSystemTests TSystemTests.cxx LIBRARIES Core)
 ROOT_ADD_GTEST(CoreCryptoRandomTest CryptoRandomTest.cxx LIBRARIES Core)
 
 ROOT_ADD_GTEST(CoreColorTests TColorTests.cxx LIBRARIES Core)
-if(CMAKE_SYSTEM_NAME MATCHES "Darwin" AND CMAKE_HOST_SYSTEM_VERSION VERSION_GREATER_EQUAL 25.4)
-  set_tests_properties(gtest-core-base-CoreColorTests PROPERTIES DISABLED True)
-  file(GENERATE
-    OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/run_until_failure
-    CONTENT
-"import subprocess
-for _ in range(100):
-    try:
-        subprocess.run(['$<TARGET_FILE:CoreColorTests>'], check=True)
-    except subprocess.CalledProcessError as e:
-        print('This test is expected to fail with a stack corruption: ', e)
-        exit(123)
-print('This test was expected to fail with a stack corruption. Check if it got fixed.')
-exit(1)"
-    TARGET CoreColorTests)
-
-  add_test(NAME CoreColor_RunUntilFailure COMMAND ${Python3_EXECUTABLE} ${CMAKE_CURRENT_BINARY_DIR}/run_until_failure)
-  set_tests_properties(CoreColor_RunUntilFailure PROPERTIES SKIP_RETURN_CODE 123)
-endif()
 
 configure_file(Foo.C Foo.C COPYONLY)
 ROOT_ADD_GTEST(IncludePathTest IncludePathTest.cxx LIBRARIES Core)

--- a/tutorials/CMakeLists.txt
+++ b/tutorials/CMakeLists.txt
@@ -739,13 +739,6 @@ foreach(t ${tutorials})
   if(${t} IN_LIST multithreaded)
     set_tests_properties(tutorial-${tname} PROPERTIES PROCESSORS ${NProcessors})
   endif()
-
-  if(CMAKE_SYSTEM_NAME MATCHES "Darwin" AND CMAKE_HOST_SYSTEM_VERSION VERSION_GREATER_EQUAL 25.4 AND tname MATCHES "gl-.*grad")
-    # ROOT-21367
-    # On mac beta, there is a stack corruption (suspected to be a compiler bug) that may fail these tests
-    set_tests_properties(tutorial-${tname} PROPERTIES DISABLED True)
-    message(NOTICE "Disabling tutorial ${tname} for mac beta")
-  endif()
 endforeach()
 
 #---Loop over all MPI tutorials and define the corresponding test---------

